From 4b49801124e3905c9709a22f748a544f9da37f00 Mon Sep 17 00:00:00 2001
From: Kevin Kong <kkong@vmware.com>
Date: Thu, 20 Aug 2020 02:55:03 +0800
Subject: [PATCH] fs/9p: fix dirty pages writeback in v9fs_evict_inode

v9fs_evict_inode should make sure the dirty pages are flushed,
then truncate inode pages and clear the inode.
---
 fs/9p/vfs_inode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/9p/vfs_inode.c b/fs/9p/vfs_inode.c
index 601a559e7d42..91b39bff8a98 100644
--- a/fs/9p/vfs_inode.c
+++ b/fs/9p/vfs_inode.c
@@ -453,9 +453,9 @@ void v9fs_evict_inode(struct inode *inode)
 {
 	struct v9fs_inode *v9inode = V9FS_I(inode);
 
+	filemap_fdatawrite(&inode->i_data);
 	truncate_inode_pages_final(&inode->i_data);
 	clear_inode(inode);
-	filemap_fdatawrite(&inode->i_data);
 
 	v9fs_cache_inode_put_cookie(inode);
 	/* clunk the fid stashed in writeback_fid */
-- 
2.20.1

